### PR TITLE
Show the unit leader in status.

### PR DESCRIPTION
**Pull request by howbazaar**
Addresses http://pad.lv/1620063.

Whether the unit is the leader or not is now returned in the FullStatus. The leader is indicated in tabular output with an * at the end of the unit name. YAML and JSON have a map value called "leader" that is only added when it is true.

The addition of the field into the FullStatus output does not need any facade version bump. It is optional, and the code handles it if it is missing by just not indicating a leader. This has been confirmed by running the new client against an old server.

## QA
```
$ juju bootstrap status-test lxd
$ juju deploy ubuntu -n 2
# wait
$ juju status
juju status
MODEL    CONTROLLER   CLOUD/REGION   VERSION
default  status-test  lxd/localhost  2.0-rc2.1

APP     VERSION  STATUS  SCALE  CHARM   STORE       REV  OS      NOTES
ubuntu  16.04    active      2  ubuntu  jujucharms    8  ubuntu  

UNIT       WORKLOAD  AGENT  MACHINE  PUBLIC-ADDRESS  PORTS  MESSAGE
ubuntu/0*  active    idle   0        10.250.171.239         ready
ubuntu/1   active    idle   1        10.250.171.225         ready

MACHINE  STATE    DNS             INS-ID         SERIES  AZ
0        started  10.250.171.239  juju-3ae63c-0  xenial  
1        started  10.250.171.225  juju-3ae63c-1  xenial  
```

### CODE REVIEW

File: apiserver/params/status.go [hunk 1]
```diff
@@ -89,6 +89,7 @@ type UnitStatus struct {
 	PublicAddress string                `json:"public-address"`
 	Charm         string                `json:"charm"`
 	Subordinates  map[string]UnitStatus `json:"subordinates"`
+	Leader        bool                  `json:"leader"`
```
Comment (axw) [inline apiserver/params/status.go]: omitempty, to save a few bytes?

File: cmd/juju/status/status_test.go [hunk 2]
```diff
@@ -1814,7 +1826,7 @@ var statusTests = []testCase{
 
 		// scoped on wordpress/0
 		scopedExpect{
-			"subordinates scoped on logging",
```
Comment (axw) [inline cmd/juju/status/status_test.go]: I think this was correct? The subordinate is scoped to the logging relation of the wordpress application?

File: cmd/juju/status/output_tabular.go [hunk 3]
```diff
@@ -177,6 +177,9 @@ func FormatTabular(writer io.Writer, forceColor bool, value interface{}) error {
 		if agentDoing != "" {
 			message = fmt.Sprintf("(%s) %s", agentDoing, message)
 		}
+		if u.Leader {
```
Comment (axw) [inline cmd/juju/status/output_tabular.go]: how about we don't display this if there's only one unit? it's noisy/uninteresting in that case.

File: cmd/juju/status/status_test.go (hunk 2 above)
Comment (howbazaar) [inline cmd/juju/status/status_test.go]: This was a copy/paste error from a previous test. Updated the comment of the test to match the args passed to status, and the comment just before the 'scopedExpect'

File: cmd/juju/status/output_tabular.go (hunk 3 above)
Comment (howbazaar) [inline cmd/juju/status/output_tabular.go]: There is no nice way to not do this. Discussed on the bug with sabdfl, and agreed that at least for the first cut, we always show it.

File: apiserver/params/status.go (hunk 1 above)
Comment (howbazaar) [inline apiserver/params/status.go]: I originally had that and took it off. Will put it back

**Comment by howbazaar** [comment]
$$merge$$

**Pull request merged**